habitat-baselines/habitat_baselines/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import pickle as pkl
+from typing import List
+
+import magnum as mn
+import numpy as np
+
+
+class Pose:
+    def __init__(self, joints, obj_transform):
+        self.joints = list(joints)
+        self.root_transform = obj_transform
+
+
+class Motion:
+    """Contains a sequential motion"""
+
+    def __init__(self, pose_array, transform_array, displacement, fps):
+        num_poses = pose_array.shape[0]
+        self.num_poses = num_poses
+        poses = []
+        for index in range(num_poses):
+            pose = Pose(
+                pose_array[index].reshape(-1),
+                mn.Matrix4(transform_array[index]),
+            )
+            poses.append(pose)
+
+        self.poses = poses
+        self.fps = fps
+        self.displacement = displacement
+
+
+class HumanoidRearrangeController:
+    """
+    Humanoid Controller, converts high level actions such as walk, or reach into joints that
+    control a URDF object.
+    """
+
+    def __init__(
+        self,
+        walk_pose_path,
+        draw_fps=60,
+        base_offset=(0, 0.9, 0),
+    ):
+        self.min_angle_turn = 0
+        self.turning_step_amount = 20
+        self.threshold_rotate_not_move = 120
+        self.base_offset = mn.Vector3(base_offset)
+
+        if not os.path.isfile(walk_pose_path):
+            raise RuntimeError(f"Path does {walk_pose_path} not exist.")
+
+        with open(walk_pose_path, "rb") as f:
+            walk_data = pkl.load(f)
+        walk_info = walk_data["walk_motion"]
+
+        self.walk_motion = Motion(
+            walk_info["joints_array"],
```

Comment:
There are a lot of magic strings in this class. Can you make them static constants ?